wikidata/client.py

```diff
@@ -177,21 +178,30 @@ def decode_datavalue(self,
                       self.datavalue_decoder)
         return decode(self, datatype, datavalue)
 
-    def request(self, path: str) -> Union[
+    def request(self, path: str) -> Optional[Union[
         bool, int, float, str,
         Mapping[
             str,
             Union[bool, int, float, str, Mapping[str, object], Sequence]
         ],
         Sequence[Union[bool, int, float, str, Mapping[str, object], Sequence]]
-    ]:
+    ]]:
         logger = logging.getLogger(__name__ + '.Client.request')
         url = urllib.parse.urljoin(self.base_url, path)
         result = self.cache_policy.get(CacheKey(url))
         if result is None:
             logger.debug('%r: no cache; make a request...', url)
-            response = self.opener.open(url)
-            buffer_ = io.TextIOWrapper(response, encoding='utf-8')
+            try:
+                response = self.opener.open(url)
+            except urllib.error.HTTPError as e:
+                logger.debug('HTTP error code: ', e.code, exc_info=True)
+                return None
+            except urllib.error.URLError as e:
+                logger.debug('Failed to reach server: ', e.reason, exc_info=True)
+                return None
```

Comment:
Intention behind this your change seems to make this method to return `None` if and only if a requested entity does not exist, right?  We should be able to distinguish server errors (Wikidata's fault) from non-existent entities.  However, catching every `HTTPError` and `URLError` could make Wikidata's temporary server errors to look like non-existent entities.

Therefore, this code should be adjusted to the following direction:

- Since `URLError` is too vague (it covers any kind of exceptions `urllib` can throw), I believe we should not catch this type of exceptions.
- There are two types of error codes in HTTP: 400 series (client errors) & 500 series (server errors).  And `HTTPError` represents both kinds of errors as I recall.  Therefore, we should not catch server errors here.  IMHO you'd better catch only exceptions with `e.code == 404` and re`raise` any others.